changelog/unreleased/pull-2773

```diff
@@ -0,0 +1,8 @@
+Enhancement: Optimize repository implementation
```

Comment:
What about "Enhancement: Optimize handling of new index entries"? That would be a bit more specific and also obsolete line 3.

Reply:
changed it.